packages/core/src/browser/navigatable.ts

```diff
@@ -87,7 +88,7 @@ export abstract class NavigatableWidgetOpenHandler<W extends NavigatableWidget>
     }
 
     protected serializeUri(uri: URI): string {
-        return uri.withoutFragment().toString();
+        return Path.normalize(uri.withoutFragment().toString());
```

Comment:
`Path.nozmalize` should be applied only top the path, i.e.
`uri.withoutFragment().withPath(Path.normalize(uri.path)).toString`

alternatively we can `normalizePath` to a URI for a shortcut, i.e. `uri.normalizePath().withoutFragment().toString()`

Reply:
actually having `normalize` on `Path` is good as well, it should return though another instance of `Path`

Reply:
> actually having `normalize` on `Path` is good as well, it should return though another instance of `Path`

Yes, I agree. I initially made `normalize` a static method which takes in a string and returns the normalized one because that's how Node.JS path implement it. But then I just realized that Node.JS path is quite different from ours.

How about making `normalize` a member function, i.e.
```typescript
normalize(): Path;
```

Reply:
yes, i like it